### ISSUE TITLE
classylss failing to build on Cori

### ISSUE THREAD

**Issue by nickhand** [bccp/conda-channel-bccp#11]
The error message is:

```
g++ -pthread -shared -L/usr/common/contrib/bccp/anaconda3/conda-bld/classylss_1502645335512/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib -Wl,-rpath=/usr/common/contrib/bccp/anaconda3/conda-bld/classylss_1502645335512/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib,--no-as-needed build/temp.linux-x86_64-2.7/classylss/binding.o -L/usr/common/contrib/bccp/anaconda3/conda-bld/classylss_1502645335512/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib -Lbuild/temp.linux-x86_64-2.7/lib -Lbuild/temp.linux-x86_64-2.7 -lclass -lgomp -lgfortran -lpython2.7 -o build/lib.linux-x86_64-2.7/classylss/binding.so -g -fPIC
/usr/bin/ld: /usr/lib64/libmvec_nonshared.a(svml_finite_alias.oS): relocation R_X86_64_PC32 against undefined symbol `_ZGVbN2v_log@@GLIBC_2.22' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
```

This seems to be due to this bug in GLIBC 2.22: https://sourceware.org/bugzilla/show_bug.cgi?id=19590

Although it looks Edison also is using GLIBC 2.22 and this builds fine on Edison. Any ideas, @rainwoodman ?

**Comment by nickhand** [comment]
I think this is because it is using the system gcc (which changed in the OS upgrade) and not the conda gcc which is what I thought was happening.

**Comment by rainwoodman** [comment]
I think adding -fPIC to the compiler flags should solve this (as the ld error stated) -- it was also the solution in the link.

**Comment by nickhand** [comment]
I think -fPIC has already been added?

**Comment by rainwoodman** [comment]
it is probably not in the install_class script?

**Comment by rainwoodman** [comment]
It is in install_class. I've filed the NERSC issue at https://nersc.service-now.com/nav_to.do?uri=incident.do%3Fsys_id=48d21b0edb2c0300ccc2f3421f961917%26sysparm_stack=incident_list.do%3Fsysparm_query=active=true

They need to update glibc to 2.23 to fix this. I am not sure how willing they are to do that. An alternative is to disable -ffastmath in the compilation flag.

**Comment by nickhand** [comment]
gcc workaround implemented in f602564
